### PR TITLE
SQLAlchemy 2 support

### PR DESCRIPTION
**Pull request by msoucy**
SQLAlchemy 2 has some nice new declarative features for writing models, but as it currently stands the code doesn't support it

### CODE REVIEW

File: signinapp/model.py [hunk 1]
```diff
@@ -107,47 +106,43 @@ def __init__(self, badge=None, owner=None):
...
+    id: Mapped[int] = mapped_column(primary_key=True)
+    username: Mapped[str] = mapped_column(unique=True)
+    name: Mapped[str]
+    preferred_name: Mapped[str | None]
+    password: Mapped[str | None]
+    subteam_id: Mapped[int | None] = mapped_column(db.ForeignKey("subteams.id"))
+    phone_number: Mapped[str | None]
+    email: Mapped[str | None]
+    address: Mapped[str | None]
+    tshirt_size: Mapped[ShirtSizes | None] = mapped_column(db.Enum(ShirtSizes))
```
Comment (msoucy) [inline signinapp/model.py]: Note to self the right hand side doesn't seem necessary per https://docs.sqlalchemy.org/en/20/orm/declarative_tables.html#using-python-enum-or-pep-586-literal-types-in-the-type-map

File: signinapp/model.py [hunk 2]
```diff
@@ -107,47 +106,43 @@ def __init__(self, badge=None, owner=None):
...
-    email = db.Column(db.String)
-    address = db.Column(db.String)
-    tshirt_size = db.Column(db.Enum(ShirtSizes))
-
-    code = db.Column(db.String, nullable=False, unique=True, default=gen_code)
-    role_id = db.Column(db.Integer, db.ForeignKey("account_types.id"))
-    approved = db.Column(db.Boolean, default=False)
-
-    stamps: list[Stamps] = db.relationship(
+    id: Mapped[int] = mapped_column(primary_key=True)
```
Comment (msoucy) [inline signinapp/model.py]: Primary key definitions can be simplified per https://docs.sqlalchemy.org/en/20/orm/declarative_tables.html#mapping-whole-column-declarations-to-python-types

File: signinapp/model.py [hunk 3]
```diff
@@ -107,47 +106,43 @@ def __init__(self, badge=None, owner=None):
...
+    password: Mapped[str | None]
+    subteam_id: Mapped[int | None] = mapped_column(db.ForeignKey("subteams.id"))
+    phone_number: Mapped[str | None]
+    email: Mapped[str | None]
+    address: Mapped[str | None]
+    tshirt_size: Mapped[ShirtSizes | None] = mapped_column(db.Enum(ShirtSizes))
+
+    code: Mapped[str] = mapped_column(unique=True, default=gen_code)
+    role_id: Mapped[int] = mapped_column(db.ForeignKey("account_types.id"))
+    approved: Mapped[bool] = mapped_column(default=False)
```
Comment (msoucy) [inline signinapp/model.py]: `NonNullBool` with a safe default value using https://docs.sqlalchemy.org/en/20/orm/declarative_tables.html#mapping-whole-column-declarations-to-python-types

File: signinapp/model.py [hunk 4]
```diff
@@ -107,47 +106,43 @@ def __init__(self, badge=None, owner=None):
...
+    awards: Mapped[list[BadgeAward]] = db.relationship(
+        back_populates="owner", cascade="all, delete-orphan"
     )
-    badges: list[Badge] = association_proxy("awards", "badge")
+    badges: AssociationProxy[list[Badge]] = association_proxy("awards", "badge")
 
     # Guardian specific data
-    guardian_user_data: Guardian = db.relationship(
-        "Guardian",
+    guardian_user_data: Mapped[Guardian] = db.relationship(
```
Comment (msoucy) [inline signinapp/model.py]: `| None`

**Review by bot190 — APPROVED** [review]

**Pull request merged**